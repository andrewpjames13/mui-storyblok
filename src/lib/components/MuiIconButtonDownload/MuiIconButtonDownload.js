import React from 'react';
import PropTypes from 'prop-types';
import { validComponentsRequired } from '../../utils/customProps';
import downloadUrl from '../../utils/downloadUrl';
import MuiIconButton from '../MuiIconButton/MuiIconButton';

/**
 * MuiIconButtonDownload is used in storyblok that will download a href
 */

export const MuiIconButtonDownload = ({
  href,
  fileName,
  iconButton,
}) => {
  const onClick = () => {
    downloadUrl(href, fileName);
  };
  const muiIconButton = iconButton[0];

  return (
    <MuiIconButton
      {...muiIconButton}
      onClick={onClick}
    />
  );
};

export default MuiIconButtonDownload;

MuiIconButtonDownload.propTypes = {
  /** url of the file you want to be downloaded */
  href: PropTypes.string.isRequired,
  /** name of file */
  fileName: PropTypes.string.isRequired,
  /** MuiIconButton Allowed maximum: 1 */
  iconButton(props, propName, componentName) {
    return validComponentsRequired(props, propName, componentName, ['MuiIconButton'], 1);
  },
};

MuiIconButtonDownload.defaultProps = {
  iconButton: [],
};
