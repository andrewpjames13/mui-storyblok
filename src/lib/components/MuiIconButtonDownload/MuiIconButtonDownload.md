MuiIconButtonDownload example:

```js
const iconButton = [{
  component: 'MuiIconButton',
  icon: [{
    component: 'MuiIcon',
    iconName: 'android',
  }],
}];

<MuiIconButtonDownload
  href="https://miro.medium.com/max/1200/1*mk1-6aYaf_Bes1E3Imhc0A.jpeg"
  fileName="cool.jpeg"
  iconButton={iconButton}
/>
```