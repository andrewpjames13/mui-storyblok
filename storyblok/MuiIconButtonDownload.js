module.exports = {
  "name": "MuiIconButtonDownload",
  "display_name": null,
  "updated_at": "2020-05-06T14:54:03.925Z",
  "schema": {
    "fileName": {
      "type": "text",
      "required": true
    },
    "href": {
      "type": "text",
      "description": "url of the file you want to be downloaded",
      "required": true
    },
    "iconButton": {
      "type": "bloks",
      "required": true,
      "maximum": "1",
      "restrict_components": true,
      "component_whitelist": [
        "MuiIconButton"
      ]
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiIconButtonDownload",
  "component_group_uuid": null
}