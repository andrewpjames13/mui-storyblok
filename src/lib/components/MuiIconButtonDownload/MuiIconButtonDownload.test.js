import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiIconButtonDownload from './MuiIconButtonDownload';

function setup() {
  const props = {
    href: 'google.com',
    fileName: 'fileName.png',
    iconButton: [{
      component: 'MuiIconButton',
      icon: [{
        component: 'MuiIcon',
        iconName: 'android',
      }],
    }],
  };

  const comp = mount(<MuiIconButtonDownload {...props} />);
  return { comp, props };
}

describe('<MuiIconButtonDownload />', () => {
  it('renders MuiIconButtonDownload', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();

    const tree = renderer.create(<MuiIconButtonDownload {...props} />);
    expect(tree).toMatchSnapshot();
  });
});

describe('clicks', () => {
  it('handleClick and call download', () => {
    const { comp } = setup();
    const { createElement } = document;
    const { createObjectURL } = window.URL;

    document.createElement = jest.fn(() => ({
      href: '',
      setAttribute: jest.fn(),
      click: jest.fn(),
      remove: jest.fn(),
    }));
    comp.find('WithStyles(ForwardRef(IconButton))').at(0).simulate('click');
    expect(document.createElement).toBeCalled();
    document.createElement = createElement;
    window.URL = createObjectURL;
  });
});
