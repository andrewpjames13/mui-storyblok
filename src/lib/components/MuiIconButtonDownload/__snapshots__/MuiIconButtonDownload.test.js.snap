// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiIconButtonDownload /> snapshot 1`] = `
<button
  className="MuiButtonBase-root MuiIconButton-root makeStyles-root-36"
  data-testid="muiIconButton"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiIconButton-label"
  >
    <span
      aria-hidden={true}
      className="material-icons MuiIcon-root makeStyles-root-37 MuiIcon-colorSecondary"
    >
      android
    </span>
  </span>
</button>
`;
